press/press/doctype/deploy_candidate/deploy_candidate.py

```diff
@@ -352,33 +344,35 @@ def schedule_build_and_deploy(
 		run_now: bool = True,
 		scheduled_time: datetime | None = None,
 	):
-		if self.status == "Scheduled":
-			return
-
+		# Here it's fine to call `self.build_and_deploy()`
+		# Since the doc has not been created yet.
 		if run_now and not is_suspended():
-			self.build_and_deploy()
-			return
+			return {"error": False, "message": self.build_and_deploy()}
 
-		self.status = "Scheduled"
-		self.scheduled_time = scheduled_time or now()
-		self.save()
-		frappe.db.commit()
-
-	def run_scheduled_build_and_deploy(self):
-		"""
-		Build and Deploy will run only if the build is Scheduled
-		and if builds have not been suspended.
-		"""
-		if self.status != "Scheduled" or is_suspended():
-			return
-		self.build_and_deploy()
+		deploy_candidate_build: DeployCandidateBuild = frappe.get_doc(
+			{
+				"doctype": "Deploy Candidate Build",
+				"deploy_candidate": self.name,
+				"no_cache": False,
+				"deploy_after_build": True,
+				"status": "Scheduled",
+				"scheduled_time": scheduled_time or now(),
+			}
+		)
+		deploy_candidate_build.insert()
+		return {"error": False, "message": deploy_candidate_build.name}
 
-	def build_and_deploy(self, no_cache: bool = False):
-		self.pre_build(
-			method="_build",
-			deploy_after_build=True,
-			no_cache=no_cache,
+	def build_and_deploy(self, no_cache: bool = False) -> str:
+		deploy_candidate_build: DeployCandidateBuild = frappe.get_doc(
+			{
+				"doctype": "Deploy Candidate Build",
+				"deploy_candidate": self.name,
+				"no_cache": no_cache,
+				"deploy_after_build": True,
+			}
 		)
+		deploy_candidate_build.insert()
+		return deploy_candidate_build.name
 
```

Comment:
Code duplication.
You probably can get rid of `build_and_deploy` function

Reply:
There is alot of cleanup in deploy candidate 😆